crates/matrix-sdk/src/test_utils/mocks/mod.rs

```diff
@@ -3243,6 +3243,244 @@ impl<'a> MockEndpoint<'a, LoginEndpoint> {
     }
 }
 
+/// A struct for building a login response with status code 200
+#[derive(Default)]
+pub struct LoginResponseBuilder200 {
+    access_token: Option<String>, // required
+    device_id: Option<String>,   // required
+    expires_in_ms: Option<u32>,
+    refresh_token: Option<String>,
+    user_id: Option<String>,    // required
+    homeserver_url: Option<String>, // required if well_known used
+    identity_url: Option<String>, // required if well_known and m.identity_server used
+}
+
+impl LoginResponseBuilder200 {
+    /// constructor for empty reponse 
+    pub fn new() -> Self { Self::default() }
+
+    /// sets access_token
+    pub fn set_access_token(&mut self, value: &str) -> &mut Self { 
+        self.access_token = Some(value.to_string()); 
+        self 
+    }
+    
+    /// sets device_id
+    pub fn set_device_id(&mut self, value: &str) -> &mut Self { 
+        self.device_id = Some(value.to_string()); 
+        self 
+    }
+    
+    /// sets expires_in_ms
+    pub fn set_expires_in_ms(&mut self, value: u32) -> &mut Self { 
+        self.expires_in_ms = Some(value); 
+        self 
+    }
+    
+    /// sets refresh_token
+    pub fn set_refresh_token(&mut self, value: &str) -> &mut Self { 
+        self.refresh_token = Some(value.to_string()); 
+        self 
+    }
+    
+    /// sets user_id
+    pub fn set_user_id(&mut self, value: &str) -> &mut Self { 
+        self.user_id = Some(value.to_string()); 
+        self 
+    }
+    
+    /// sets homeserver_url
+    pub fn set_homeserver_url(&mut self, value: &str) -> &mut Self { 
+        self.homeserver_url = Some(value.to_string()); 
+        self 
+    }
+    
+    /// sets identity_url
+    pub fn set_identity_url(&mut self, value: &str) -> &mut Self { 
+        self.identity_url = Some(value.to_string()); 
+        self 
+    }
+    
+    /// builds a ResponseTemplate from builder
+    pub fn build(&mut self) -> ResponseTemplate {
```

Comment:
I think the `build()` method won't be necessary once the constructor takes all the required fields.

This would make the template consistent with the other custom templates we have.